static/js/components/SourceTableFilterForm.jsx

```diff
@@ -123,20 +141,102 @@ const SourceTableFilterForm = ({ handleFilterSubmit }) => {
     []
   );
 
-  const { handleSubmit, register, control, reset } = useForm();
+  const gcnEvents = useSelector((state) => state.gcnEvents);
+  const [selectedGcnEventId, setSelectedGcnEventId] = useState(null);
+
+  useEffect(() => {
+    if (gcnEvents?.length > 0 || !gcnEvents) {
+      dispatch(gcnEventsActions.fetchGcnEvents());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const { handleSubmit, register, control, reset, getValues } = useForm();
 
   const handleClickReset = () => {
     reset();
   };
 
+  const gcnEventsLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  gcnEvents?.events.forEach((gcnEvent) => {
+    gcnEventsLookUp[gcnEvent.id] = gcnEvent;
+  });
+
+  const gcnEventsSelect = gcnEvents
+    ? [
+        {
+          id: -1,
+          dateobs: "Clear Selection",
+        },
+        ...gcnEvents.events,
+      ]
+    : [];
+
+  const validate = (formData) => {
+    let valid = true;
+    if (formData.gcneventid !== "" || formData.localizationid !== "") {
+      if (formData.startDate === "" || formData.endDate === "") {
+        dispatch(
+          showNotification(
+            "Please enter First and Last Detected dates when filtering by GCN Event",
+            "error"
+          )
+        );
+        valid = false;
+      } else if (new Date(formData.startDate) > new Date(formData.endDate)) {
+        dispatch(
+          showNotification(
+            "First Detected date must be before Last Detected date",
+            "error"
+          )
+        );
+        valid = false;
+      } // check if there are more than 31 days between start and end date
+      else if (
+        (new Date(formData.endDate) - new Date(formData.startDate)) /
+          (1000 * 60 * 60 * 24) >
+        31
+      ) {
+        dispatch(
+          showNotification(
+            "Please enter a date range less than 31 days",
+            "error"
```

Comment:
Done ! What i did on the frontend is that it uses the value coming from the config, but to avoid errors, if this value is not accessible for some reason, it defaults to 31.

Reply:
Done ! What i did on the frontend is that it uses the value coming from the config, but to avoid errors, if this value is not accessible for some reason, it defaults to 31.